cosmo_tester/test_suites/bootstrap_based_tests/cli_test.py

```diff
@@ -33,117 +33,100 @@
     get_resource_path,
     sh_bake,
 )
+from cosmo_tester.framework.test_hosts import get_latest_manager_image_name
 
 WINRM_PORT = 5985
 
 
 @pytest.fixture(scope='function')
-def cli_package_tester(ssh_key, attributes, tmpdir, logger):
-    logger.info('Using temp dir: %s', tmpdir)
-    tmpdir = Path(tmpdir)
-
-    tf_inputs = get_terraform_inputs(attributes, ssh_key)
-    tester = _CliPackageTester(tmpdir, tf_inputs, ssh_key, logger)
-
-    yield tester
-
-    tester.perform_cleanup()
-
-
-@pytest.fixture(scope='function')
-def windows_cli_package_tester(ssh_key, attributes, tmpdir, logger):
-    logger.info('Using temp dir: %s', tmpdir)
-    tmpdir = Path(tmpdir)
-
-    tf_inputs = get_terraform_inputs(attributes, ssh_key)
-    tester = _WindowsCliPackageTester(tmpdir, tf_inputs, ssh_key, logger)
-
-    yield tester
-
-    tester.perform_cleanup()
-
+def package_tester(request, ssh_key, attributes, tmpdir, logger):
+    _package_tester_mapping = {
+        'linux': _CliPackageTester,
+        'osx': _OSXCliPackageTester,
+        'windows': _WindowsCliPackageTester
+    }
+    platform = request.param
+    tester_class = _package_tester_mapping[platform]
 
-@pytest.fixture(scope='function')
-def osx_cli_package_tester(ssh_key, attributes, tmpdir, logger):
     logger.info('Using temp dir: %s', tmpdir)
     tmpdir = Path(tmpdir)
 
-    tf_inputs = get_terraform_inputs(attributes, ssh_key)
-    tester = _OSXCliPackageTester(tmpdir, tf_inputs, ssh_key, logger)
+    tester = tester_class(tmpdir, attributes, ssh_key, logger)
 
     yield tester
 
     tester.perform_cleanup()
 
 
-def test_cli_on_centos_7(cli_package_tester, attributes):
-    cli_package_tester.inputs.update({
+@pytest.mark.parametrize('package_tester', ['linux'], indirect=True)
+def test_cli_on_centos_7(package_tester, attributes):
```

Comment:
I wonder if we could parameterise this based on OS name and then provide a dict in the file which mapped out the inputs for each- that'd mean keeping one dict that contained just the extra inputs and then having one test runner that used pytest's parameterisation to pull the inputs from that dict for each OS in turn.